AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift

```diff
@@ -66,10 +66,14 @@ public class PersistentHitQueue: HitQueuing {
             let semaphore = DispatchSemaphore(value: 0)
             self.processor.processHit(entity: hit, completion: { [weak self] success in
                 if success {
-                    // successful processing of hit, remove it from the queue, move to next hit
-                    _ = self?.dataQueue.remove()
-
-                    self?.processNextHit()
+                    // successful processing of hit
+                    // attempt to remove it from the queue and process next hit if successful
+                    if self?.dataQueue.remove() ?? false {
+                        self?.processNextHit()
+                    } else {
+                        // deleting the hit from the database failed
+                        // need to delete the database to try and recover
```

Comment:
can we leave a warning log in the else part here?

Reply:
@sbenedicadb  should we create a Github issue for it, then we can fix it in a future version?

Reply:
@yangyansong-adbe #670